go-controller/pkg/cni/ovs.go

```diff
@@ -110,66 +111,86 @@ func ofctlExec(args ...string) (string, error) {
-		// Function checks for OpenFlow flows to know the pod is ready
-		// TODO(trozet): in the future use a more stable mechanism provided by OVN:
-		// https://bugzilla.redhat.com/show_bug.cgi?id=1839102
-
-		// ensure the OVS interface is still active. It may have been cleared by a subsequent CNI ADD
-		// and if so, there's no need to keep checking for flows
-		out, err := ovsGet("Interface", ifaceName, "external-ids", "iface-id")
-		if err != nil {
-			return false, nil
-		}
-		if out != ifaceID {
-			return false, fmt.Errorf("OVS sandbox port %s is no longer active (probably due to a subsequent "+
-				"CNI ADD)", ifaceName)
-		}
-
-		// Query the flows by mac address for in_port_security
-		queries := []query{
-			{
-				match:  "dl_src=" + mac,
-				tables: []int{9},
-			},
+	// Query the flows by mac address for in_port_security
+	queries := []query{
+		{
+			match:  "dl_src=" + mac,
+			tables: []int{9},
+		},
+	}
+	for _, ifAddr := range ifAddrs {
+		var ipMatch string
+		if !utilnet.IsIPv6(ifAddr.IP) {
+			ipMatch = "ip,ip_dst"
+		} else {
+			ipMatch = "ipv6,ipv6_dst"
 		}
-		for _, ifAddr := range ifAddrs {
-			var ipMatch string
-			if !utilnet.IsIPv6(ifAddr.IP) {
-				ipMatch = "ip,ip_dst"
-			} else {
-				ipMatch = "ipv6,ipv6_dst"
+		// add queries for out_port_security
+		// note we need to support table 48 for 20.06 OVN backwards compatibility. Table 49 is now
+		// where out_port_security lives
+		queries = append(queries,
+			query{fmt.Sprintf("%s=%s", ipMatch, ifAddr.IP), []int{48, 49}},
+		)
+	}
+	for _, query := range queries {
+		found := false
+		// Look for a match in any table to be considered success
+		for _, table := range query.tables {
+			queryStr := fmt.Sprintf("table=%d,%s", table, query.match)
+			// ovs-ofctl dumps error on stderr, so stdout will only dump flow data if matches the query.
+			stdout, err := ofctlExec("dump-flows", "br-int", queryStr)
+			if err == nil && len(stdout) > 0 {
+				found = true
+				break
 			}
-			// add queries for out_port_security
-			// note we need to support table 48 for 20.06 OVN backwards compatibility. Table 49 is now
-			// where out_port_security lives
-			queries = append(queries,
-				query{fmt.Sprintf("%s=%s", ipMatch, ifAddr.IP), []int{48, 49}},
-			)
 		}
-		for _, query := range queries {
-			found := false
-			// Look for a match in any table to be considered success
-			for _, table := range query.tables {
-				queryStr := fmt.Sprintf("table=%d,%s", table, query.match)
-				// ovs-ofctl dumps error on stderr, so stdout will only dump flow data if matches the query.
-				stdout, err := ofctlExec("dump-flows", "br-int", queryStr)
-				if err == nil && len(stdout) > 0 {
-					found = true
-					break
-				}
+		if !found {
+			return false, nil
+		}
+	}
+
+	return true, nil
+}
+
+func waitForPodFlows(mac string, ifAddrs []*net.IPNet, ifaceName, ifaceID string, ctx context.Context) error {
```

Comment:
The pattern is to pass context as first argument

> Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it. The Context should be the first parameter, typically named ctx:

https://golang.org/pkg/context/